code/core/system/proxy-velodyne16/src/ProxyVelodyne16.cpp

```diff
@@ -17,51 +17,65 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#include <iostream>
 #include <fstream>
+#include <iostream>
 #include <memory>
 
+#include "ProxyVelodyne16.h"
+#include "opendavinci/odcore/base/KeyValueConfiguration.h"
 #include "opendavinci/odcore/data/Container.h"
 #include "opendavinci/odcore/io/conference/ContainerConference.h"
 #include "opendavinci/odcore/wrapper/SharedMemoryFactory.h"
-#include "ProxyVelodyne16.h"
 
 
 namespace opendlv {
 namespace core {
 namespace system {
 namespace proxy {
 
-    using namespace odcore::wrapper;
-    using namespace odcore::io::udp;
+using namespace odcore::wrapper;
+using namespace odcore::io::udp;
 
-    ProxyVelodyne16::ProxyVelodyne16(const int32_t &argc, char **argv) :
-        TimeTriggeredConferenceClientModule(argc, argv, "ProxyVelodyne16"),
-        VelodyneSharedMemory(SharedMemoryFactory::createSharedMemory(NAME, SIZE)),
-        udpreceiver(UDPFactory::createUDPReceiver(RECEIVER, PORT)),
-        v16d(VelodyneSharedMemory,getConference()){}
+ProxyVelodyne16::ProxyVelodyne16(const int32_t &argc, char **argv)
+    : TimeTriggeredConferenceClientModule(argc, argv, "ProxyVelodyne16")
+    , memoryName()
+    , memorySize(0)
+    , udpReceiverIP()
+    , udpPort(0)
+    , VelodyneSharedMemory(NULL)
+    , udpreceiver(NULL)
+    , v16dSp(NULL) {}
 
-    ProxyVelodyne16::~ProxyVelodyne16() {}
+ProxyVelodyne16::~ProxyVelodyne16() {}
 
-    void ProxyVelodyne16::setUp() {
-        udpreceiver->setStringListener(&v16d);
-        // Start receiving bytes.
-        udpreceiver->start();
-    }
+void ProxyVelodyne16::setUp() {
+    memoryName = getKeyValueConfiguration().getValue< string >("ProxyVelodyne16.sharedMemory.name");
```

Comment:
Please adjust the module's name here (and in the following) to proxy-velodyne16).